simulator/src/main/java/com/hazelcast/simulator/worker/metronome/SimpleMetronome.java

```diff
@@ -36,6 +36,21 @@ public static Metronome withFixedIntervalMs(int intervalMs) {
         return new SimpleMetronome(TimeUnit.MILLISECONDS.toNanos(intervalMs));
     }
 
+    /**
+     * Creates a {@link Metronome} instance with a fixed frequency in Hz.
+     *
+     * @param frequency frequency
+     * @return a {@link Metronome} instance
+     */
+    public static Metronome withFixedFrequency(float frequency) {
+        if (frequency == 0) {
+            return EMPTY_METRONOME;
```

Comment:
Hmm, right. An `IllegalArgumentException` would be the better fit, wdyt? In our use case `0 Hz` would mean, that we would wait forever, which is most likely never wanted.